cypress/plugins/nodeManager/plugin.ts

```diff
@@ -217,6 +229,11 @@ class Node {
       this.logger.log("ignoring stop node command, node wasn't running");
     }
   }
+
+  private cleanup(): void {
+    this.logger.log("cleaning up state");
+    fs.removeSync(this.storagePath);
```

Comment:
See comment about blocking code above

Reply:
This is called in the exit hook, which AFAIK should only do `sync` things. Should we then switch to: https://www.npmjs.com/package/async-exit-hook?

Reply:
From what I can see this is only called in the `exit` handler of a process and not in the `exitHook()` callback. It is ok for the `exit` event handler to be async (and in fact, it already is) and await `cleanup()`.

When it comes to `exitHook()` it turns out that the `exit` event on a the node `process` is never triggered when we kall `stopAllNodes()` in `exitHook()`. I don’t think this is too big of an issue if we occasionally leak these temporary directories.

Reply:
> From what I can see this is only called in the exit handler of a process and not in the exitHook() callback. It is ok for the exit event handler to be async (and in fact, it already is) and await cleanup().

Let there be `async` then: eda5044f75097aa8ceefffbdd6baa3e7084269a0.

> When it comes to exitHook() it turns out that the exit event on a the node process is never triggered when we kall stopAllNodes() in exitHook(). I don’t think this is too big of an issue if we occasionally leak these temporary directories.

You're right, the child process exit handlers didn't get called because we killed the child process with `.kill`. Sending a shutdown signal should let the child processes clean up after themselves: 9eec21f550be4b4a033be67f6ecf74650b8accf0.

Reply:
> Sending a shutdown signal lets the child processes clean up after themselves: 9eec21f.

Ugh, didn't work the way I thought it would, never mind.